src/components/wasd-controls.js

```diff
@@ -42,27 +47,29 @@ module.exports.Component = registerComponent('wasd-controls', {
   },
 
   tick: function (time, delta) {
+    var currentPosition;
     var data = this.data;
     var el = this.el;
     var movementVector;
-    var position;
+    var position = this.position;
     var velocity = this.velocity;
 
-    // Use seconds.
-    delta = delta / 1000;
+    if (!velocity[data.adAxis] && !velocity[data.wsAxis] &&
+        isEmptyObject(this.keys)) { return; }
 
-    // Get velocity.
+    // Update velocity.
+    delta = delta / 1000;
     this.updateVelocity(delta);
+
     if (!velocity[data.adAxis] && !velocity[data.wsAxis]) { return; }
 
     // Get movement vector and translate position.
+    currentPosition = el.object3D.position;
```

Comment:
I would keep using `getAttribute` to keep using the public API instead of coupling with the internals. I don't think there's any difference.